File: 5. Sliding Windows/424. Longest Repeating Character Replacement/Solution.cpp
#include <string>
#include <unordered_map>

class Solution {
public:
    int characterReplacement(std::string s, int k) {
        int res = 0;
        std::unordered_map<char, int> freqMap;
        int max_freq = 0;
        int left = 0;

        for (int right = 0; right < s.size(); ++right)
        {
            ++freqMap[s[right]];
            max_freq = std::max(freqMap[s[right]], max_freq);

            if (right - left + 1 - max_freq > k)
            {
                --freqMap[s[left]];
                ++left;
            }

            res = right - left + 1;
        }

        return res;
    }
};

File: 5. Sliding Windows/424. Longest Repeating Character Replacement/Solution.py
from collections import Counter

class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        freq_map = Counter()
        max_freq = 0
        count = 0
        left = 0
        right = 0

        for right in range(len(s)):
            freq_map[s[right]] += 1
            max_freq = max_freq if max_freq > freq_map[s[right]] else freq_map[s[right]]

            if right - left + 1 - max_freq > k:
                freq_map[s[left]] -= 1
                left += 1

            count = right - left + 1 

        return count